core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroup.java

```diff
@@ -291,7 +290,17 @@ private static boolean hasChanges(List<Endpoint> oldEndpoints, List<Endpoint> ne
         return false;
     }
 
-    private void completeInitialEndpointsFuture(List<Endpoint> endpoints) {
+    @Override
+    protected List<Endpoint> latestValue() {
```

Comment:
Do we need this method now?

Reply:
Not for correctness. It was overridden to provide an optimized implementation.

Reply:
Isn't caliing ` whenReady().isDone()` is more optimized than accessing the volatile `endpoints` field when it's not initialized?

Reply:
If `isDone()` is true, it would access two volatile fields.

Reply:
Ah, that is true. 👍